corehq/blobs/migrate.py

```diff
@@ -411,6 +522,50 @@ def _get_progress_logger(self):
         return CouchProcessorProgressLogger(self.doc_types)
 
 
+class SqlMigrator(Migrator):
+
+    def __init__(self, slug, reindexer, doc_migrator_class):
+        types = [reindexer.model_class]
+        assert not hasattr(types[0], "get_db"), types[0] # not a couch model
+        doc_migrator = partial(doc_migrator_class, blob_helper=reindexer.blob_helper)
+        super(SqlMigrator, self).__init__(slug, types, doc_migrator)
+        self.reindexer = reindexer
+
+    def migrate(self, *args, **kw):
+        with allow_form_processing_queries():
+            return super(SqlMigrator, self).migrate(*args, **kw)
+
+    def _get_document_provider(self):
+        return SqlDocumentProvider(self.iteration_key, self.reindexer)
+
+
+class MultiDbMigrator(object):
+
+    def __init__(self, slug, couch_types, sql_reindexers, doc_migrator_class):
+        self.slug = slug
+        self.migrators = migrators = []
+        def db_key(doc_type):
+            if isinstance(doc_type, tuple):
+                doc_type = doc_type[1]
+            return doc_type.get_db().dbname
+        for key, types in groupby(sorted(couch_types, key=db_key), key=db_key):
+            migrators.append(Migrator(slug, list(types), doc_migrator_class))
+        for rex in sql_reindexers:
+            migrators.append(SqlMigrator(slug, rex(), doc_migrator_class))
+
+    def migrate(self, filename, *args, **kw):
+        def filen(n):
+            return None if filename is None else "{}.{}".format(filename, n)
+        migrated = 0
+        skipped = 0
+        for n, item in enumerate(self.migrators):
+            one_migrated, one_skipped = item.migrate(filen(n), *args, **kw)
+            migrated += one_migrated
+            skipped += one_skipped
+            print("\n")
+        return migrated, skipped
```

Comment:
Note to self: need to wait to write `BlobMigrationState` until all sub-migrations are complete. It's currently written after each sub-migration, which works but is not correct.